auth0/v3/authentication/token_verifier.py

```diff
@@ -0,0 +1,225 @@
+import json
+import time
+
+import jwt
+import requests
+
+from auth0.v3.exceptions import TokenValidationError
+
+
+class SignatureVerifier():
+    DISABLE_JWT_CHECKS = {
+        "verify_signature": True,
+        "verify_exp": False,
+        "verify_nbf": False,
+        "verify_iat": False,
+        "verify_aud": False,
+        "verify_iss": False,
+        "require_exp": False,
+        "require_iat": False,
+        "require_nbf": False,
+    }
+
+    def __init__(self, algorithm):
+        if not algorithm or type(algorithm) != str:
+            raise ValueError("algorithm must be specified.")
+        self._algorithm = algorithm
+
+    def _fetch_key(self, key_id=None):
+        raise NotImplementedError
```

Comment:
will be implemented by the subclasses, providing "keys" AKA shared secrets or public certificates